e2e/test/scenarios/sharing/public-dashboard.cy.spec.js

```diff
@@ -286,6 +288,54 @@ describe("scenarios > public > dashboard", () => {
     cy.findByText("Registerkarte als PDF exportieren").should("be.visible");
     cy.url().should("include", "locale=de");
   });
+
+  it("should respect click behavior", () => {
+    createDashboardWithQuestions({
+      dashboardName: "test click behavior",
+      questions: [
+        {
+          name: "orders",
+          query: {
+            "source-table": ORDERS_ID,
+            limit: 5,
+          },
+        },
+      ],
+      cards: [
+        {
+          visualization_settings: {
+            column_settings: {
+              '["name","TOTAL"]': {
+                click_behavior: {
+                  type: "link",
+                  linkType: "url",
+                  linkTemplate: "https://metabase.com",
+                },
+              },
+            },
+          },
+        },
+      ],
+    }).then(({ dashboard }) => {
+      visitPublicDashboard(dashboard.id);
+    });
+
+    // This is a hacky way to intercept the link click we create an a element
+    // with href on fly and remove it afterwards in lib/dom.js
+    cy.window().then(win => {
+      cy.spy(win.document.body, "appendChild").as("appendChild");
+    });
```

Comment:
I assume we're actually navigating to `metabase.com` here after the click action though, so I wonder if we could just assert that the URL has changed to `metabase.com`?

Reply:
The link it's opened on a new tab, cypress doesn't handle tabs multiple tabs.
The way we usually handle these tests it by checking if the url has the _blank attribute but in this case, the `a` tag is generated on the fly when clicking on the link by `openInBlankWindow` function

Reply:
@npretto I see, that is tricky indeed! Then spying on [clickLink](https://github.com/metabase/metabase/pull/50067#discussion_r1846103558) as you mentioned makes sense.